modules/core/src/main/scala/gem/ProgramId.scala

```diff
@@ -0,0 +1,211 @@
+   * This is the catch-all type for otherwise unparseable ids, so it is guaranteed that the string
+   * representation of a `Nonstandard` via `.format` is *not* parseable in to a standard science or
+   * daily program id. This data type has no public constructor and no `.copy` method, as these
+   * could violate the above invariant. The only way to get an instance is via `.fromString`.
+   */
+  sealed abstract case class Nonstandard(
+    override val siteOption:        Option[Site],
+    override val semesterOption:    Option[Semester],
+    override val programTypeOption: Option[ProgramType],
+                 tail:              String
+  ) extends ProgramId(
+    siteOption,
+    semesterOption,
+    programTypeOption
+  ) {
+    override def format =
+      Nonstandard.format(siteOption, semesterOption, programTypeOption, tail)
+
+  }
+  object Nonstandard {
+
+    /** Format the components of a `Nonstandard`. */
+    def format(
+      siteOption:        Option[Site],
+      semesterOption:    Option[Semester],
+      programTypeOption: Option[ProgramType],
+      tail:              String
+    ): String =
+      List(
+        siteOption       .map(_.shortName).toList,
+        semesterOption   .map(_.format)   .toList,
+        programTypeOption.map(_.shortName).toList,
+        List(tail)
+      ).flatten.intercalate("-")
+
+    /**
+     * Parse a `Nonstandard` program id from a string, if possible. Note that this will fail if
+     * the the `s` represents a valid science or daily program id.
+     */
+    def fromString(s: String): Option[Nonstandard] =
+      ProgramId.fromString(s) collect {
+        case id: Nonstandard => id
+      }
+
+    /** `Nonstandard` program ids are ordered pairwise by their data members. */
+    implicit val NonStandardOrder: Order[Nonstandard] =
+      Order[Option[Site]]       .contramap[Nonstandard](_.siteOption)        |+|
+      Order[Option[Semester]]   .contramap[Nonstandard](_.semesterOption)    |+|
+      Order[Option[ProgramType]].contramap[Nonstandard](_.programTypeOption) |+|
+      Order[String]             .contramap[Nonstandard](_.tail)
+
+  }
+
+  /** Parse a `ProgramId` from string, if possible. */
+  def fromString(s: String): Option[ProgramId] =
+    Science.fromString(s) orElse
+    Daily  .fromString(s) orElse
+    Parsers.parseExact(Parsers.programId.nonstandard)(s).map {
+      case (os, om, op, t) => new Nonstandard(os, om, op, t) {}
+    }
+
+  /**
+   * Programs are ordered lexically by prodict prefix (Daily, Nonstandard, then Science) and then
+   * by the defined orderings for individual cases when constructors match.
+   */
+  implicit val ProgramIdOrder: Order[ProgramId] =
+    Order.order {
+      case (a: Science,     b: Science)     => a cmp b
+      case (a: Daily,       b: Daily)       => a cmp b
+      case (a: Nonstandard, b: Nonstandard) => a cmp b
+      case (a,              b)              => a.productPrefix cmp b.productPrefix
```

Comment:
Yeah, I'm not really happy with that either. I'll write out the cases. Should we order by Science, Daily, then Nonstandard?